### ISSUE TITLE
Pagination Bug in Query Gutenberg Block

### ISSUE THREAD

**Issue by webprom** [WordPress/gutenberg#36382]
### Description

First item in paginated blog custom query doesn't have a link to page 1 when you're on next pages. I.e. ?query-15-page=6 wouldn't have link in pagination to ?query-15-page=1. 

`<a class="page-numbers" href="">1</a>`

Previous and Next links work ok.

### Step-by-step reproduction instructions

1 )  Insert Query block and set a custom query, i.e. with some specific category and with enough pages
2 )  Add Pagination sub block with next and previous + page numbers
3 )  Check the link on link number 1, when you're on page 2 or any next page 

### Screenshots, screen recording, code snippet

![image](https://user-images.githubusercontent.com/3159046/141183076-6813b3c1-eac1-42b5-aa50-96348108f287.png)
![image](https://user-images.githubusercontent.com/3159046/141183245-26d5b16e-47fa-49d4-a819-29b70012135d.png)

### Environment info

- Latest WP  5.8.1 WITHOUT Gutenberg Plugin
- Custom Theme

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
I am not able to reproduce this with Gutenberg trunk active:
```
<div class="wp-block-query-pagination-numbers"><a class="page-numbers" href="http://localhost:8889/">1</a>
<span aria-current="page" class="page-numbers current">2</span>
<a class="page-numbers" href="http://localhost:8889/page/3/">3</a>
<a class="page-numbers" href="http://localhost:8889/page/4/">4</a>
<a class="page-numbers" href="http://localhost:8889/page/5/">5</a></div>
```

Nor with WordPress 5.9 alpha without Gutenberg active:
```
<div class="wp-block-query-pagination-numbers"><a class="page-numbers" href="http://localhost:8889/">1</a>
<span aria-current="page" class="page-numbers current">2</span>
<a class="page-numbers" href="http://localhost:8889/page/3/">3</a>
<a class="page-numbers" href="http://localhost:8889/page/4/">4</a>
<a class="page-numbers" href="http://localhost:8889/page/5/">5</a></div>
```

**Comment by webprom** [comment]
Upgraded to 5.8.2 but the issue is still there. 
Again, no Gutenberg plugin, and a custom query.
Switched to theme Twenty Twenty-One and disabled all plugins again.
Permalinks structure:  `/%category%/article/%year%/%monthnum%/%day%/%postname%/`

![image](https://user-images.githubusercontent.com/3159046/141290854-ac5710f7-84da-491d-8eeb-896c292129f4.png)


```
<div class="wp-block-query-pagination"><a href="/about-us/blog/?query-15-page=2" class="wp-block-query-pagination-previous">« Previous Page</a>

<div class="wp-block-query-pagination-numbers"><a class="page-numbers" href="">1</a>
<a class="page-numbers" href="?query-15-page=2">2</a>
<span aria-current="page" class="page-numbers current">3</span>
<a class="page-numbers" href="?query-15-page=4">4</a>
<a class="page-numbers" href="?query-15-page=5">5</a>
<span class="page-numbers dots">…</span>
<a class="page-numbers" href="?query-15-page=121">121</a></div>

<a href="/about-us/blog/?query-15-page=4" class="wp-block-query-pagination-next">Next Page »</a></div>
```

**Comment by webprom** [comment]
But with the Gutenberg plugin installed it works. Was it fixed in 5.9 alpha?

```
<div class="wp-container-618d02f4d9ff6 wp-block-query-pagination">
<a href="/innovation/awards/?query-15-page=1&amp;cst" class="wp-block-query-pagination-previous">Previous Page</a>

<div class="wp-block-query-pagination-numbers"><a class="page-numbers" href="?cst">1</a>
<span aria-current="page" class="page-numbers current">2</span>
<a class="page-numbers" href="?query-15-page=3&amp;cst">3</a></div>

<a href="/innovation/awards/?query-15-page=3&amp;cst" class="wp-block-query-pagination-next">Next Page</a>
</div>
```

**Comment by carolinan** [comment]
I have retested this today with WordPress 5.9.0 with and without Gutenberg and it is working. I am closing the issue, and it can be re-opened if the problem re-appears.
